core/lib/core/studies/study_edit.ex

```diff
@@ -162,6 +172,14 @@ defmodule Core.Studies.StudyEdit do
     struct(Core.Studies.StudyEdit, opts)
   end
 
+  defp put_default(map, key, value) do
+    if !Map.has_key?(map, key) || map[key] === nil do
```

Comment:
A possible shorter alternative could be: `Map.update(map, key, value, & &1||value)`